src/types/pgduckdb_enum.cpp

```diff
@@ -0,0 +1,140 @@
+#include "pgduckdb/types/pgduckdb_enum.hpp"
+#include "pgduckdb/pgduckdb_utils.hpp"
+#include "duckdb/common/extra_type_info.hpp"
+
+namespace pgduckdb {
+
+using duckdb::idx_t;
+using duckdb::LogicalTypeId;
+using duckdb::PhysicalType;
+
+// This is medium hacky, we create extra space in the Vector that would normally only hold the strings corresponding to
+// the enum's members. In that extra space we store the enum member oids.
+
+// We do this so we can find the OID from the offset (which is what DuckDB stores in a Vector of type ENUM) and return
+// that when converting the result from DuckDB -> Postgres
+
+LogicalType
+PGDuckDBEnum::CreateEnumType(std::vector<HeapTuple> &enum_members) {
```

Comment:
I think it makes sense to merge this function `ConvertPostgresEnumToDuckEnum`. They are so strongly coupled that I don't think it has any benefit to separate them like this. i.e. `ConvertPostgresEnumToDuckEnum` is the only caller of this function and that this takes a vector of HeapTuples, which should already be sorted. If we'd keep them separate that requirement on already sorted should be documented in this function its comment, but if we merge the functions such a comment is not necessary.